projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitFormFragment.java

```diff
@@ -0,0 +1,124 @@
+package hasadna.noloan2;
+
+import android.app.DatePickerDialog;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.TextView;
+import java.util.Calendar;
+
+import hasadna.noloan2.protobuf.LawsuitProto.Lawsuit;
+import noloan.R;
+
+public class LawsuitFormFragment extends Fragment {
+  private static final String TAG = "LawsuitFormFragment";
+
+  EditText receivedDate;
+  DatePickerDialog datePickerDialog;
+  Calendar calendar;
+  Button buttonNext;
+  TextView selectCourt;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+  }
+
+  @Override
+  public View onCreateView(
+      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    ViewGroup rootView =
+        (ViewGroup) inflater.inflate(R.layout.lawsuit_fragment_form, container, false);
+
+    receivedDate = rootView.findViewById(R.id.EditText_receivedSpamDate);
+    selectCourt = rootView.findViewById(R.id.textView_selectCourt_FormFragment);
+    buttonNext = rootView.findViewById(R.id.button_formFragment_next);
+
+    // If court was already selected, display its name
+    if (((LawsuitActivity) getActivity()).selectedCourt != null) {
+      selectCourt.setText(
+          "בית המשפט לתביעות קטנות - " + ((LawsuitActivity) getActivity()).selectedCourt.getName());
+    }
+    selectCourt.setOnClickListener(v -> ((LawsuitActivity) getActivity()).displayCourtPicker(this));
```

Comment:
we need to make some of the fields requairment to continue, such as court.
then you will not need the CourtPicker in the Activity and can move it to here,

Reply:
>we need to make some of the fields requairment to continue, such as court.

I don't understand, what do you mean by "continue"?

Reply:
Making some field requirement to move to the next page, if they are not filled you cannot click "next"

Reply:
I think if you want to file a lawsuit, not necessarily you have all the required information at one moment.

Either way the selection of the court isn't part of the generated PDF itself...

I think letting the users experience and feel they have some progress done, and letting them actually see and hold a formatted PDF of their lawsuit on their device, is more "motivating" to continue on and progressing to do some further "homework" if needed, and filling the missing fields.

I agree with what you say about having required field limitation. And maybe even before that, validation for the fields that are already there. I think though it's something to have further thoughts about.

I'm thinking for the bottom line result:

If you limit the users - Some will take off.
Out of those who took off, some would have continued if they have reached some "success" level. And even if not, it wouldn't change if you had limited them or not. (Tell me what you think..?)

In this case, it will just change their "last" memory from the usage of the app.
One with a PDF on their device, and the other from giving up :) 

IMO holding a PDF in your device leaves the door open and raises up the potential for continuing the progress.

Those who won't take off... 
If with limitation - have the required fields, OR will put the effort in getting this information at the same moment. (As for now, the form doesn't hold the last state if opening the app again).

If with no limitation - fields stays the same if they had them, and so does the effort they were willing to give in case they had some missing. I think even more if they experience watching the lawsuit on their device.